### PR TITLE
Stop using strfime format as fallback

### PR DESCRIPTION
**Pull request by evanxg852000**
Stop accepting any non-recognised format as `strftime` format and check if it contains special strftime markers.
Closes https://github.com/quickwit-oss/quickwit/issues/2479

### CODE REVIEW

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs [hunk 1]
```diff
@@ -323,4 +351,32 @@ mod tests {
...
+        for format in formats {
+            let error_str = DateTimeInputFormat::from_str(format)
+                .unwrap_err()
+                .to_string();
+            assert_eq!(error_str, format!("Unknown input format: `{format}`."));
+        }
+    }
+
+    #[test]
+    fn test_fail_date_time_output_format_from_str_with_unknown_format() {
```
Comment (fmassot) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs:373]: there is one redundant test I think.

Comment (evanxg852000) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs:373]: Sorry, I don't understand. 
There is the DateTimeInputFormat and DateTimeOutputFormat with a different error message for this purpose.
Also DateTimeInputFormat has the   `unix_timestamp_*` set of formats

Do you mean only testing one is enough?

Comment (fmassot) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs:373]: My mistake, sorry for the bad comment!

Comment (evanxg852000) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs:373]: No worries :+1:

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs [hunk 2]
```diff
@@ -143,7 +161,12 @@ impl FromStr for DateTimeInputFormat {
             "rfc2822" => DateTimeInputFormat::RFC2822,
             "rfc3339" => DateTimeInputFormat::RCF3339,
             "unix_timestamp" => DateTimeInputFormat::Timestamp,
-            _ => DateTimeInputFormat::Strptime(StrptimeParser::from_str(date_time_format_str)?),
+            _ => {
+                if !is_strftime_formatting(date_time_format_str) {
+                    return Err(format!("Unknown input format: `{date_time_format_str}`."));
```
Comment (fmassot) [inline quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_format.rs:166]:
```suggestion
                    return Err(format!("Unknown input format: `{date_time_format_str}`. A custom date time format must contain at least one `strftime` special characters ."));
```

**Review by fmassot — APPROVED** [review]

**Pull request merged**